docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -292,3 +322,12 @@ Components not decided yet at current phase are not included in these views.
 ### Key Scenario: tenant & DNS provisioning
 
 ![tenant & DNS provision](images/network-scenario-tenant-init.png)
+
+### Activity Flow: Pod - from creation to running
+
+![Pod-activity-diag](images/pod-activity-diagram.png)
+
+1. It is strongly recommended to annotate new pods with arktos.futurewei.com/network-readiness as hint for scheduler. It is not mandatory, though - kubelet and cni plugin will have to cope with it and gets it right eventually. This kind of initial pod annotation can be done by a custom admission control.
```

Comment:
Not sure if this "network-readiness" annotation get couple with scheduling.  Some network providers may need to know the scheduling decision before it can initialize pod network.  So there could be a deadlock. (I know flannel is not a network provider type here, but Flannel is an example that it allocates IP based on node selection.)

Reply:
this network-readiness is for scheduler to go after network controller has done necessary preparation, which unblock scheduling. If network controller needs to do some post-scheduled task, it can still go by watching pod spec.nodeName and do whatever it needs to.